Re-use existing paramiko connection in SSHHook instead of recreating

rootcss opened the pull request:
SSHHook was creating the paramiko client every time execute() method was called. This PR makes change to re-use the previously established connection.

closes: https://github.com/apache/airflow/issues/10874

kaxil reviewed (changes requested):
tests are failing
```
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_no_intermediate_dir_error_get
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_with_intermediate_dir_error_get
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_with_intermediate_dir_put
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_json_file_transfer_get
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_json_file_transfer_put
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_pickle_file_transfer_get
ERROR tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_pickle_file_transfer_put
FAILED tests/providers/amazon/aws/transfers/test_s3_to_sftp.py::TestS3ToSFTPOperator::test_s3_to_sftp_operation
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_no_intermediate_dir_error_get
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_no_intermediate_dir_error_put
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_with_intermediate_dir_error_get
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_file_transfer_with_intermediate_dir_put
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_json_file_transfer_get
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_json_file_transfer_put
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_pickle_file_transfer_get
FAILED tests/providers/sftp/operators/test_sftp.py::TestSFTPOperator::test_pickle_file_transfer_put
= 9 failed, 6663 passed, 294 skipped, 2 xfailed, 2 xpassed, 14 warnings, 7 errors in 1857.85s (0:30:57) =
```

coopergillan committed:
Add type annotations to ZendeskHook, update unit test (#10888)

* Add type annotations to ZendeskHook

__What__

* Add correct type annotations to ZendeskHook and each method
* Update one unit test to call an empty dictionary rather than a
NoneType since the argument should be a dictionary

__Why__

* Building out type annotations is good for the code base
* The query parameter is accessed with an index at one point, which
means that it cannot be a None type, but should rather be defaulted to
an empty dictionary if not provided

* Remove useless return
